lstchain/reco/utils.py

```diff
@@ -778,6 +781,127 @@ def get_geomagnetic_delta(zen, az, geomag_dec=None, geomag_inc=None, time=None):
     return delta
 
 
+def get_intensity_threshold(data):
+    """
+    Obtain from a dl2 data table the peak of the differential
+    intensity (I) spectrum, i.e. dR/dI = dN/dt/dI (events/s/p.e.), and
+    determine from it the intensity at which the 50% of the peak rate 
+    is reached on the rising edge.
+    
+    Parameters
+    ----------
+    data: pandas DataFrame or astropy.table.QTable of dl2 events.
+
+    Returns
+    -------
+    xmax: (float) Intensity (in p.e.) at which the peak of dR/dI is reached 
+    
+    ymax: (float) Value of dR/dI at peak (events/s/p.e.)
+
+    x50: (float) Intensity (in p.e.) at which 50% ofthe peak of dR/dI is reached
+    on the rising edge of the peak. Will be nan if not found in the search range
+
+    y50: (float) 50% of the peak dR/dI (events/s/p.e.). Will be nan if not found
+    in the search range
+
+    bincenters: (array) the centers (in log10(intensity)) of the bins of 
+    the intensity histogram used in the calculation (p.e.)
+
+    nevents: (array) the bin contents of the histogram used in the calculation
+    (events/s/p.e.)
+    
+    """
+    
+    # p.e. range where to look for the "true" cosmics peak
+    min_intensity = 25 
+    max_intensity = 1000  
+    # At lower intensity there are occasioanally fake peaks from stars, 
+    # satellites, meteors...
+    # At higher intensity there are sometimes peak from external light 
+    # sources, and from mis-tagged FF events
+    step = 0.02 # in log10(intensity), for histogramming & peak finding
+    
+    efftime, _ = get_effective_time(data)
+
+    nbins = 1 + int( (np.log10(max_intensity) - np.log10(min_intensity)) / step)
+    bins = np.logspace(np.log10(min_intensity), 
+                       np.log10(max_intensity), nbins)
+    binwidth = np.diff(bins)
+    nevents, _ = np.histogram(data['intensity'], bins=bins)
+    nevents = nevents.astype('float') / (binwidth*efftime.to_value(u.s))
```

Comment:
why changing the value of a variable here instead of defining a new one?

Reply:
Right, the name was also misleading. Now it is drdi